src/OrbitGl/PagefaultTrack.cpp

```diff
@@ -0,0 +1,137 @@
+          this, time_graph, viewport, layout, "Minor Pagefault Track", series_names, capture_data,
+          indentation_level + 1)} {
+  SetName("Pagefault Track");
+  SetLabel("Pagefault Track");
+}
+
+float PagefaultTrack::GetHeight() const {
+  if (collapse_toggle_->IsCollapsed()) {
+    return major_pagefault_track_->GetHeight();
+  }
+
+  float height = layout_->GetTrackTabHeight();
+  if (!major_pagefault_track_->IsEmpty()) {
+    height += major_pagefault_track_->GetHeight() + layout_->GetSpaceBetweenSubtracks();
+  }
+  if (!minor_pagefault_track_->IsEmpty()) {
+    height += minor_pagefault_track_->GetHeight() + layout_->GetSpaceBetweenSubtracks();
+  }
+  return height;
+}
+
+std::vector<orbit_gl::CaptureViewElement*> PagefaultTrack::GetVisibleChildren() {
+  std::vector<CaptureViewElement*> result;
+  if (collapse_toggle_->IsCollapsed()) return result;
+
+  if (!major_pagefault_track_->IsEmpty()) result.push_back(major_pagefault_track_.get());
+  if (!minor_pagefault_track_->IsEmpty()) result.push_back(minor_pagefault_track_.get());
+  return result;
+}
+
+void PagefaultTrack::Draw(Batcher& batcher, TextRenderer& text_renderer,
+                          uint64_t current_mouse_time_ns, PickingMode picking_mode,
+                          float z_offset) {
+  float track_height = GetHeight();
+  float track_width = viewport_->GetVisibleWorldWidth();
+
+  SetPos(viewport_->GetWorldTopLeft()[0], pos_[1]);
+  SetSize(track_width, track_height);
+
+  UpdatePositionOfSubtracks();
+
+  Track::Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode, z_offset);
+
+  if (collapse_toggle_->IsCollapsed()) return;
+
+  if (!major_pagefault_track_->IsEmpty()) {
+    major_pagefault_track_->SetSize(viewport_->GetVisibleWorldWidth(),
+                                    major_pagefault_track_->GetHeight());
+    major_pagefault_track_->Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode,
+                                 z_offset);
+  }
+
+  if (!minor_pagefault_track_->IsEmpty()) {
+    minor_pagefault_track_->SetSize(viewport_->GetVisibleWorldWidth(),
+                                    minor_pagefault_track_->GetHeight());
+    minor_pagefault_track_->Draw(batcher, text_renderer, current_mouse_time_ns, picking_mode,
+                                 z_offset);
+  }
+}
+
+void PagefaultTrack::UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t max_tick,
+                                      PickingMode picking_mode, float z_offset) {
+  UpdatePositionOfSubtracks();
+
+  if (!major_pagefault_track_->IsEmpty()) {
+    major_pagefault_track_->UpdatePrimitives(batcher, min_tick, max_tick, picking_mode, z_offset);
+  }
+
+  if (collapse_toggle_->IsCollapsed()) return;
+
+  if (!minor_pagefault_track_->IsEmpty()) {
+    minor_pagefault_track_->UpdatePrimitives(batcher, min_tick, max_tick, picking_mode, z_offset);
+  }
+}
+
+void PagefaultTrack::UpdatePositionOfSubtracks() {
```

Comment:
Or also could be here the indentation level

Reply:
I'm thinking that indentation level should affect the position of the subtracks. Or am I wrong?